src/api/core/two_factor/yubikey.rs

```diff
@@ -145,15 +145,23 @@ async fn activate_yubikey(data: Json<EnableYubikeyData>, headers: Headers, mut c
 
     // Ensure they are valid OTPs
     for yubikey in &yubikeys {
-        if yubikey.len() == 12 {
-            // YubiKey ID
+        if yubikey.is_empty() || yubikey.len() == 12 {
             continue;
         }
 
         verify_yubikey_otp(yubikey.to_owned()).await.map_res("Invalid Yubikey OTP provided")?;
     }
 
-    let yubikey_ids: Vec<String> = yubikeys.into_iter().map(|x| (x[..12]).to_owned()).collect();
+    let yubikey_ids: Vec<String> = yubikeys
+        .into_iter()
+        .filter_map(|x| {
+            if x.len() >= 12 {
+                Some((x[..12]).to_owned())
+            } else {
+                None
+            }
+        })
```

Comment:
Wouldn't Rust 1.50 be easier?
```suggestion
        .filter_map(|x| (x.len() >= 12).then(|| x[..12].to_owned()))
```

Reply:
Ah! Yes, that might look more nice indeed. But i was wondering why you were mentioning a very old Rust version hehe.

Thanks for the pointer.

Reply:
Another option here might also be to use .get() with a range, which returns an optional value instead of panicking, which I think is a bit clear to understand?
```rust
    let yubikey_ids: Vec<String> = yubikeys.into_iter().filter_map(|x| x.get(..12)).map(str::to_owned).collect();
```

Reply:
That looks a bit neater indeed.
Adjusted :).